File: Almost Sorted.cpp
#include <bits/stdc++.h>

using namespace std;

bool isSorted(vector<int> &arr){
    for(int i=1;i<arr.size();i++)
        if(arr[i]<arr[i-1]) return false;
    return true;
}

int main()
{
    int n;
    cin >> n;
    vector<int> arr(n);
    for(int i=0;i<n;i++)
        cin>>arr[i];
    if(isSorted(arr)){
        cout<<"yes";
        return 0;
    }
    int a,b;
    for(a=0;a<arr.size() && arr[a]<=arr[a+1];a++);
    for(b=arr.size()-1;a>=0 && arr[b]>=arr[b-1];b--);
    int aa=a+1,bb=b+1;
    swap(arr[a++],arr[b--]);
    if(isSorted(arr)){
        cout<<"yes\nswap "<<aa<<" "<<bb;
        return 0;
    }
    while(a<b) swap(arr[a++],arr[b--]);
    if(isSorted(arr)){
        cout<<"yes\nreverse "<<aa<<" "<<bb;
        return 0;
    }
    cout<<"no";
}

File: Almost Sorted.py
_ = input()
arr = list(map(int,input().split()))
sarr = sorted(arr)
us=[]
if sarr==arr:
    print('yes')
else:
    for i in range(len(sarr)):
        if sarr[i]!=arr[i]:
            us.append(i)
    if len(us)==2:
        print(f"yes\nswap {us[0]+1} {us[-1]+1}")
    elif arr[us[0]:us[-1]+1] == sorted(arr[us[0]:us[-1]+1],reverse=True):
        print(f"yes\nreverse {us[0]+1} {us[-1]+1}")
    else:
        print("no")
